Parsers/ASimDns/ProductParsers/ASimDnsMicrosoftSysmon.yaml

```diff
@@ -15,88 +15,88 @@ References:
 ParserName: ASimDnsMicrosoftSysmon
 ParserQuery: | 
-  let RCodeTable=datatable(DnsResponseCode:int,DnsResponseCodeName:string)[
-    // See https://docs.microsoft.com/windows/win32/debug/system-error-codes--9000-11999-
-     0, 'NOERROR'
-   , 9001, "FORMERR"
-   , 9002,"SERVFAIL"
-   , 9003,'NXDOMAIN'
-   , 9004,'NOTIMP'
-   , 9005,'REFUSED'
-   , 9006,'YXDOMAIN'
-   , 9007,'YXRRSET'
-   , 9008,'NXRRSET'
-   , 9009,'NOTAUTH'
-   , 9010,'NOTZONE'
-   , 9011,'DSOTYPENI'
-   , 9016,'BADVERS'
-   , 9016,'BADSIG'
-   , 9017,'BADKEY'
-   , 9018,'BADTIME'
-   , 9019,'BADMODE'
-   , 9020,'BADNAME'
-   , 9021,'BADALG'
-   , 9022,'BADTRUNC'
-   , 9023,'BADCOOKIE'
-   , 1460, 'TIMEOUT'
-  ];
-  let ParsedDnsEvent_Event =() {
-    Event 
-    | where Source == "Microsoft-Windows-Sysmon" and EventID==22
-    | parse EventData with 
-      * 
-      '<Data Name="RuleName">' RuleName:string '</Data>' 
-      '<Data Name="UtcTime">' EventEndTime:datetime '</Data>'
-      '<Data Name="ProcessGuid">{' ProcessGuid:string '}</Data>'
-      '<Data Name="ProcessId">' ProcessId:string '</Data>'
-      '<Data Name="QueryName">' DnsQuery:string '</Data>'
-      '<Data Name="QueryStatus">' DnsResponseCode:int  '</Data>'
-      '<Data Name="QueryResults">' DnsResponseName:string '</Data>'
-      '<Data Name="Image">' Process:string '</Data>'
-      *
-      | project-rename SrcUsername = UserName
-      | extend Username = SrcUsername
-      | project-away EventData, ParameterXml, RenderedDescription  
+    let RCodeTable=datatable(DnsResponseCode:int,DnsResponseCodeName:string)[
+        // See https://docs.microsoft.com/windows/win32/debug/system-error-codes--9000-11999-
+        0, 'NOERROR'
+      , 9001, "FORMERR"
+      , 9002,"SERVFAIL"
+      , 9003,'NXDOMAIN'
+      , 9004,'NOTIMP'
+      , 9005,'REFUSED'
+      , 9006,'YXDOMAIN'
+      , 9007,'YXRRSET'
+      , 9008,'NXRRSET'
+      , 9009,'NOTAUTH'
+      , 9010,'NOTZONE'
+      , 9011,'DSOTYPENI'
+      , 9016,'BADVERS'
+      , 9016,'BADSIG'
+      , 9017,'BADKEY'
+      , 9018,'BADTIME'
+      , 9019,'BADMODE'
+      , 9020,'BADNAME'
+      , 9021,'BADALG'
+      , 9022,'BADTRUNC'
+      , 9023,'BADCOOKIE'
+      , 1460, 'TIMEOUT'
+      ];
+    let ParsedDnsEvent_Event =() {
+        Event 
+        | where Source == "Microsoft-Windows-Sysmon" and EventID==22
+        | parse EventData with 
+            * 
+            '<Data Name="RuleName">' RuleName:string '</Data>' 
+            '<Data Name="UtcTime">' EventEndTime:datetime '</Data>'
+            '<Data Name="ProcessGuid">{' ProcessGuid:string '}</Data>'
+            '<Data Name="ProcessId">' ProcessId:string '</Data>'
+            '<Data Name="QueryName">' DnsQuery:string '</Data>'
+            '<Data Name="QueryStatus">' DnsResponseCode:int  '</Data>'
+            '<Data Name="QueryResults">' DnsResponseName:string '</Data>'
+            '<Data Name="Image">' Process:string '</Data>'
+            '<Data Name="User">'SrcUsername:string '</Data>' *
```

Comment:
You got the backward compatibility OK later on (though I had comments there as well). In this case I think you missed. If there is no user field, the parse will fail entirely. User has to be extracted in a separate parse or extract call. Probably true later on as well.

Reply:
this is the first version for this parser